pkg/generators/models/models.go

```diff
@@ -144,78 +147,114 @@ func (m *Model) Render(ctx context.Context, writer io.Writer) error {
+
 	if out == nil {
-		out = &openapi3.SchemaRef{
-			Value: &openapi3.Schema{},
-		}
+		out = &openapi3.SchemaRef{}
+	}
+	if out.Value == nil {
+		out.Value = &openapi3.Schema{}
+	}
+
+	if right == nil {
+		right = &openapi3.SchemaRef{}
 	}
+	if right.Value == nil {
+		right.Value = &openapi3.Schema{}
+	}
+
 	if right.Ref != "" {
 		out.Ref = right.Ref
 	}
 	if right.Value.Type != "" {
 		out.Value.Type = right.Value.Type
 	}
+
+	// merge docs
 	if right.Value.Title != "" {
 		out.Value.Title = right.Value.Title
 	}
 	if right.Value.Description != "" {
 		out.Value.Description = right.Value.Description
 	}
-	if right.Value.Format != "" {
-		out.Value.Format = right.Value.Format
-	}
-	if len(out.Value.OneOf) > 0 {
-		out.Value.OneOf = append(out.Value.OneOf, right.Value.OneOf...)
+	if right.Value.ExternalDocs != nil {
+		out.Value.ExternalDocs = right.Value.ExternalDocs
 	}
-	if len(out.Value.AllOf) > 0 {
-		out.Value.AllOf = append(out.Value.AllOf, right.Value.AllOf...)
+	if right.Value.Example != nil {
+		out.Value.Example = right.Value.Example
 	}
-	if len(out.Value.AnyOf) > 0 {
-		out.Value.AnyOf = append(out.Value.AnyOf, right.Value.AnyOf...)
+
+	// override main type properties
+	if right.Value.Format != "" {
+		out.Value.Format = right.Value.Format
 	}
-	if right.Value.AdditionalProperties != nil {
-		out.Value.AdditionalProperties = mergeSchemas(out.Value.AdditionalProperties, right.Value.AdditionalProperties)
+	if right.Value.Nullable != out.Value.Nullable {
+		out.Value.Nullable = true
 	}
 	if right.Value.AdditionalPropertiesAllowed != nil {
 		out.Value.AdditionalPropertiesAllowed = right.Value.AdditionalPropertiesAllowed
 	}
-	if right.Value.Nullable {
-		out.Value.Nullable = true
-	}
-	if len(right.Value.Required) > 0 {
-		out.Value.Required = append(out.Value.Required, right.Value.Required...)
-	}
-	if len(right.Value.Enum) > 0 {
-		out.Value.Enum = append(out.Value.Enum, right.Value.Enum...)
+
+	out.Value.OneOf = append(out.Value.OneOf, right.Value.OneOf...)
+	out.Value.AllOf = append(out.Value.AllOf, right.Value.AllOf...)
+	out.Value.AnyOf = append(out.Value.AnyOf, right.Value.AnyOf...)
+	out.Value.Enum = append(out.Value.Enum, right.Value.Enum...)
+	out.Value.Required = append(out.Value.Required, right.Value.Required...)
+
+	if right.Value.AdditionalProperties != nil {
+		out.Value.AdditionalProperties = deepMerge(
+			out.Value.AdditionalProperties,
+			right.Value.AdditionalProperties,
+			nil, // here we merge all over, without `passed`
+		)
 	}
+
 	if len(right.Value.Properties) > 0 {
-		if out.Value == nil {
-			out.Value = &openapi3.Schema{}
-		}
 		if out.Value.Properties == nil {
 			out.Value.Properties = make(map[string]*openapi3.SchemaRef)
 		}
 		for k, v := range right.Value.Properties {
-			out.Value.Properties[k] = v
+			out.Value.Properties[k] = resolveAllOf(v, passed)
 		}
 	}
+
+	if right.Value.Type == "array" && right.Value.Items != nil {
+		out.Value.Items = resolveAllOf(right.Value.Items, passed)
+	}
```

Comment:
That's the actual fix I was looking for: this recursively resolves `allOf` for array items too.